extensions-core/parquet-extensions/src/main/java/org/apache/druid/data/input/parquet/simple/ParquetGroupFlattenerMaker.java

```
@@ -0,0 +1,126 @@
+{
+  private static final MappingProvider DEFAULT_MAPPING_PROVIDER = new MappingProvider()
+  {
+    @Override
+    public <T> T map(Object source, Class<T> targetType, Configuration configuration)
+    {
+      throw new NotImplementedException();
+    }
+
+    @Override
+    public <T> T map(Object source, TypeRef<T> targetType, Configuration configuration)
+    {
+      throw new NotImplementedException();
+    }
+  };
+
+  private final Configuration jsonPathConfiguration;
+  private final ParquetGroupConverter converter;
+
+  ParquetGroupFlattenerMaker(boolean binaryAsString)
+  {
+    this.converter = new ParquetGroupConverter(binaryAsString);
+    this.jsonPathConfiguration = Configuration.builder()
+                                              .jsonProvider(new ParquetGroupJsonProvider(converter))
+                                              .mappingProvider(DEFAULT_MAPPING_PROVIDER)
+                                              .options(EnumSet.of(Option.SUPPRESS_EXCEPTIONS))
+                                              .build();
+  }
+
+  @Override
+  public Set<String> discoverRootFields(Group obj)
+  {
+    return obj.getType()
+              .getFields()
+              .stream()
+              .filter(Type::isPrimitive)
+              .map(Type::getName)
+              .collect(Collectors.toSet());
+  }
+
+  @Override
+  public Object getRootField(Group obj, String key)
+  {
+    Object val = converter.convertField(obj, key);
+    return finalizeConversion(val);
+  }
+
+  @Override
+  public Function<Group, Object> makeJsonPathExtractor(String expr)
+  {
+    final JsonPath jsonPath = JsonPath.compile(expr);
+    return record -> {
+      Object val = jsonPath.read(record, jsonPathConfiguration);
+      return finalizeConversion(val);
+    };
+  }
+
+  @Nullable
+  @Override
+  public Function<Group, Object> makeJsonQueryExtractor(String expr)
+  {
+    return null;
```

I think it'd be better to throw an exception here